### PR TITLE
Support configurable timeout

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description
- added configurable timeout

### Issues Resolved
https://github.com/opensearch-project/reporting-cli/issues/61

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rupal-bq** [comment]
CI failure not related to this change.

File: src/arguments.js [hunk 1]
```diff
@@ -67,7 +67,10 @@ async function getCommandArguments() {
             .env(ENV_VAR.EMAIL_NOTE))
         .addOption(new Option('--selfsignedcerts <flag>', 'enable or disable self-signed certicates for smtp transport')
             .default(DEFAULT_SELF_SIGNED_CERTIFICATES)
-            .choices(['true', 'false']));
+            .choices(['true', 'false']))
+        .addOption(new Option('--timeout <timeout>', 'timeout for generating report')
```
Comment (joshuali925) [inline src/arguments.js:71]: it might be better to include the unit in the option name (`--timeout-ms`) or in the description

**Review by joshuali925 — COMMENTED** [review]
also update the user guide?

File: USER_GUIDE.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ Option | Default Value | Valid Options |  Environment Variable | Description
 --subject | This is an email containing your dashboard report | - | OPENSEARCH_SUBJECT | subject for the email
 --note | Hi,\nHere is the latest report! | string or path to text file | OPENSEARCH_EMAIL_NOTE | The email body
 --selfsignedcerts | false | true, false | - | enable or disable self-signed certicates for smtp transport
+--timeout | 300000ms | - | OPENSEARCH_TIMEOUT | timeout for generating report in ms
```
Comment (joshuali925) [inline USER_GUIDE.md:44]: the value shouldn't contain `ms` or does it not matter?

Comment (rupal-bq) [reply USER_GUIDE.md]: Good catch. the value shouldn't contain `ms`.

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**